anvil/tests/it/otterscan.rs

```diff
@@ -0,0 +1,307 @@
+
+    assert_eq!(res0.number, Some(0.into()));
+    assert_eq!(res1.number, Some(1.into()));
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn can_call_ots_get_api_level() {
+    let (api, _handle) = spawn(NodeConfig::test()).await;
+
+    assert_eq!(api.ots_get_api_level().await.unwrap(), 8);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn can_call_ots_has_code() {
+    let (api, handle) = spawn(NodeConfig::test()).await;
+    let provider = handle.http_provider();
+
+    let wallet = handle.dev_wallets().next().unwrap();
+    let sender = wallet.address();
+    let client = Arc::new(SignerMiddleware::new(provider, wallet));
+    api.mine_one().await;
+
+    let mut deploy_tx = MulticallContract::deploy(Arc::clone(&client), ()).unwrap().deployer.tx;
+    deploy_tx.set_nonce(0);
+
+    let pending_contract_address = get_contract_address(sender, deploy_tx.nonce().unwrap());
+
+    // no code in the address before deploying
+    assert!(!api
+        .ots_has_code(pending_contract_address, BlockNumber::Number(1.into()))
+        .await
+        .unwrap());
+
+    client.send_transaction(deploy_tx, None).await.unwrap();
+
+    let num = client.get_block_number().await.unwrap();
+    // code is detected after deploying
+    assert!(api.ots_has_code(pending_contract_address, BlockNumber::Number(num)).await.unwrap());
+
+    // code is not detected for the previous block
+    assert!(!api
+        .ots_has_code(pending_contract_address, BlockNumber::Number(num - 1))
+        .await
+        .unwrap());
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn can_call_ots_get_transactionn_error() {
+    let prj = TempProject::dapptools().unwrap();
+    prj.add_source(
+        "Contract",
+        r#"
+pragma solidity 0.8.13;
+contract Contract {
+    uint256 public number;
+
+    function setNumber(uint256 num) public {
+        require(num != 0, "RevertStringFooBar");
+        number = num;
+    }
+}
+"#,
+    )
+    .unwrap();
+
+    let mut compiled = prj.compile().unwrap();
+    assert!(!compiled.has_compiler_errors());
+    let contract = compiled.remove_first("Contract").unwrap();
+    let (abi, bytecode, _) = contract.into_contract_bytecode().into_parts();
+
+    let (api, handle) = spawn(NodeConfig::test()).await;
+    let provider = handle.ws_provider().await;
+
+    let wallet = handle.dev_wallets().next().unwrap();
+    let client = Arc::new(SignerMiddleware::new(provider, wallet));
+
+    // deploy successfully
+    let factory = ContractFactory::new(abi.clone().unwrap(), bytecode.unwrap(), client);
+    let contract = factory.deploy(()).unwrap().send().await.unwrap();
+
+    let call = contract.method::<_, ()>("setNumber", U256::zero()).unwrap();
+    let resp = call.send().await;
+    dbg!(&resp);
+
+    // TODO: Resp is a Result<PendingTransaction. How can I force it to be included in a block?
+    // api.mine_one().await will still give out a block with no txs
+    // resp.unwrap() fails because the tx reverts
```

Comment:
@Evalir maybe you have some pointers here?
I was scanning other tests (particularly `anvil/tests/it/revert.rs`) but couldn't find an example of how to force a failed transaction to be mined

Reply:
oh hey had not seen this—they should be mined? looping in @mattsse here

Reply:
solved with help from mattsse